src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -843,15 +856,44 @@ public InterfaceConfig getInterfaceConfig(String fullName) {
    */
   public SingleResourceNameConfig getSingleResourceNameConfig(String entityName) {
     ResourceNameConfig resourceNameConfig = getResourceNameConfigs().get(entityName);
-    if (resourceNameConfig != null && resourceNameConfig instanceof SingleResourceNameConfig) {
+    if (resourceNameConfig instanceof SingleResourceNameConfig) {
       return (SingleResourceNameConfig) resourceNameConfig;
     }
-    if (resourceNameConfig != null && resourceNameConfig instanceof ResourceNameOneofConfig) {
+    if (resourceNameConfig instanceof ResourceNameOneofConfig) {
       ResourceNameOneofConfig oneofConfig = (ResourceNameOneofConfig) resourceNameConfig;
       if (Iterables.size(oneofConfig.getSingleResourceNameConfigs()) > 0) {
         return Iterables.get(oneofConfig.getSingleResourceNameConfigs(), 0);
       }
     }
     return null;
   }
+
+  // Describes if a GAPIC config was given as input. This is a wrapper around checking
+  // `if (ConfigProto == null)`, and letting ConfigProto be non-null internally avoids NPEs.
+  public enum GapicConfigPresence {
```

Comment:
Two questions, not sure of correct answer for either:
- Should this have its own file?
- Is an enum necessary? Or could/should this be a boolean?